docker/install/dnf/Dockerfile

```diff
@@ -0,0 +1,56 @@
+# This Dockerfile enables an iterative development workflow where you can make
+# a change and test it out quickly. The majority of commands in this file will
+# be cached, making the feedback loop typically quite short. The workflow is
+# as follows:
+#   1. Set up pre-conditions for the system in puppet code using `deploy.pp`.
+#   2. Make a change to the module.
+#   3. Run `./docker/bin/install.sh rocky` from the project directory.
+#   4. Review the output. Repeat steps 2-3 as needed.
+#
+# At the end of execution, you will see a line like:
+# Dependencies resolved.
+# ========================================================================================================================================
+#  Package                          Architecture               Version                                  Repository                   Size
+# ========================================================================================================================================
+# Installing:
+#  puppet-agent                     x86_64                     8.10.0-1.el8                             puppet8                      27 M
+
+ARG BASE_IMAGE=rocky:8
+FROM ${BASE_IMAGE}
+
+# Use this to force a cache reset (e.g. for output purposes)
+#COPY $0 /tmp/Dockerfile
+
+# Install some other dependencies for ease of life.
+RUN  dnf update -y \
+  && dnf install -y git \
+  && dnf clean all
+
+# This is also duplicated in the docker/bin/helpers/run-upgrade.sh.
+ENV module_path=/tmp/modules
+WORKDIR "${module_path}/puppet_agent"
+COPY metadata.json ./
+
+# Installing dependencies from source. These versions should be within the range
+# of `dependencies` in metadata.json.
+RUN git clone --depth 1 https://github.com/puppetlabs/puppetlabs-stdlib ../stdlib --branch v9.7.0
```

Comment:
Is it a good idea to hardcode versions here? Or should it maybe checkout the latest tag?

Reply:
I'm doing this for each module, let me know if there's a better way? 

```
RUN git clone --tags https://github.com/puppetlabs/puppetlabs-stdlib ../stdlib && \
    $(cd ../stdlib && git checkout $(git describe --tags $(git rev-list --tags --max-count=1)))
```

Reply:
I'm by far no git expert, but that looks good to me.